gips/0065-subgraph-availability-manager.md

```diff
@@ -0,0 +1,123 @@
+---
+
+# Abstract
+
+This GIP proposes increasing the decentralization in The Graph by introducing a new smart contract `SubgraphAvailabilityManager`.
+
+# Motivation
+
+The Subgraph Availability Oracle (SAO) plays a crucial role in The Graph's ecosystem. Its main function is to verify the availability and validity of subgraph files. Currently there is a single instance of the SAO in operation executing open-source code accessible at [The Graph's Subgraph Oracle repository](https://github.com/graphprotocol/subgraph-oracle). This SAO instance runs at five-minute intervals, performing a series of checks to assess the state of subgraph files. In instances where the SAO identifies unavailability or invalidity in a subgraph's files, it sends a transaction to the `RewardsManager` contract to deny indexing rewards for that subgraph.
+
+The proposed change aims to enhance the decentralization of the protocol by introducing five separate oracles operators and requiring a minimum of three votes among them before any decisive action is taken. All oracles operators will be bound by the **Subgraph Availability Oracle Operator Charter** included in this GIP which outlines their roles, responsibilities, and operational guidelines.
+
+# High-Level Description
+
+The new smart contract `SubgraphAvailabilityManager` will be exclusively deployed to the Arbitrum One network and will function as a mediator between five distinct oracle operators and the `RewardsManager` contract. It will feature a vote aggregation system and enforce an execution threshold requiring three votes to alter the state of a subgraph deployment within the `RewardsManager` contract. Additionally, it will incorporate a `voteTimeLimit` timeframe to monitor current votes, after which they expire and become invalid.
+
+It's important to note that this system operates effectively only if a minimum of three oracles are honest and running. The system will be able to report subgraph availability whenever three or more of the oracle instances are running correctly. According to the Charter guidelines, all oracles are expected to operate in compliance with established standards. In the event that any oracles are found not to be performing up to these standards, measures will be taken to replace them, ensuring the system's integrity and continuous effective operation.
+
+The `SubgraphAvailabilityManager` contract will be governed by the council. While the contract itself will not be upgradeable, a new contract can be deployed, and the council can set the `RewardsManager` oracle address to the new deployment. The council will have the authority to add or remove oracles operators and update the `voteTimeLimit` timeframe.
+
+# Detailed Specification
+
+The number of oracle operators will be strictly enforced as five. Upon contract initialization, five valid accounts must be provided, and each oracle must maintain their index position, crucial for passing it as a parameter when casting votes.
```

Comment:
Any reason why enforcing only five operators? Would it be possible to make it at least five with a cap limit?

Reply:
We're enforcing five operators because that's what we're launching with. We decided having fixed values for number of oracles and threshold was safer than having maximum and updating. This doesn't mean that in the future we can't have more than five oracles though, we can deploy a new contract with new configurations.